design-documents/static-checks-for-templates.md

```diff
@@ -0,0 +1,27 @@
+### Overview
+
+At the time of this writing only 85% of the `*.phtml` files in Magento core are being statically 
+tested due to `@codingStandardsIgnore` present in over 700 templates. There is a current security initiative 
+that requires a large number of templates to be modified and they need to be covered by static checks moving 
+forward as part of the Acceptance Criteria of the initiate. 
+
+However, due to the lack of current static template checks the codebase has thousands of static template errors and will require many 
+hours to address every issue in every file. Additionally, many of the tests are not reasonable, checking incorrectly, or simply gratuitous for `*.phtml` templates. 
+
+
+### Design
+
+To address this issue for the immediate efforts as well as for all developers in the future I am proposing to remove the following sniffs
+for `*.phtml` files.
+
+|Sniff|Rationale|
+|-----|---------|
+| `Magento2.Files.LineLength` |HTML is frequently much cleaner when kept on single lines even if they are longer. This sniff also incorrectly counts lines containing PHP.|
```

Comment:
Could you please explain what does `Squiz.WhiteSpace.ScopeClosingBrace` check? Do you propose completely remove `PEAR.ControlStructures.ControlSignature`? I think alternative syntax is more readable.

Reply:
It works fine for regular php files where it asserts that control structures look like this:
```php
if ($foo) {
  //something
}
```

However for phtml files it gives different results based on the context but essentially something like this:

```php
<foo <?php if ($foo) { ?>checked<?php } ?>
<foo <?php if ($foo) : ?>checked<?php endif; ?>
```

is required to be turned into

```php
<foo <?php if ($foo) { 
          ?>checked<?php } ?>
<foo <?php if ($foo) : 
          ?>checked<?php endif; ?>
```

or some other weird format.

Reply:
phpcbf automatically convert it to this at the moment.

Reply:
@michaelyu0123 Yes I understand but the purpose was for code to be cleaner, this is messy.

Reply:
So `PEAR.ControlStructures.ControlSignature` requires to add new line? Then I'm ok with removing it. Does `Squiz.WhiteSpace.ScopeClosingBrace` check that whitespace exist or doesn't exist before closing tag?

I think that this syntax
`<?php if ($foo): ?><h2>Hello world</h2><?php endif ?>`
or
```
<?php if ($foo): ?>
    <h2>Hello world</h2>
<?php endif ?>
```
both are good examples. Do we preserve this style?

Reply:
@melnikovi 

> So PEAR.ControlStructures.ControlSignature requires to add new line?

Yes, and only for some cases so it's inconsistent. 

> Does Squiz.WhiteSpace.ScopeClosingBrace check that whitespace exist...

There are different checks for proper (non-newline) whitespace. 

> Do we preserve this style?

Yes, that's the idea for the PR is that I think we should preserve these styles